pkg/util/sets/ipnet.go

```diff
@@ -0,0 +1,119 @@
+/*
+Copyright 2016 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package sets
+
+import (
+	"net"
+	"strings"
+)
+
+type IPNet map[string]*net.IPNet
```

Comment:
I think we should move it to a different pkg with a big comment to fold it
back into autogenerated sets.

What do you mean that IPNet can't be used as a map key?

http://play.golang.org/p/xLDdbzvUUS

On Tue, Mar 1, 2016 at 5:07 PM, Justin Santa Barbara <
notifications@github.com> wrote:

> In pkg/util/sets/ipnet.go
> <https://github.com/kubernetes/kubernetes/pull/21907#discussion_r54664583>
> :
>
> > +
> > +Unless required by applicable law or agreed to in writing, software
> > +distributed under the License is distributed on an "AS IS" BASIS,
> > +WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
> > +See the License for the specific language governing permissions and
> > +limitations under the License.
> > +*/
> > +
> > +package sets
> > +
> > +import (
> > +	"net"
> > +	"strings"
> > +)
> > +
> > +type IPNet map[string]*net.IPNet
>
> I think it is consistent, though not feature complete. But I'm probably
> missing something?
>
> Can we make a decision here? This PR is blocking a bunch of other PRs and
> that set of PRs are probably my only blockers for cutting the branch as far
> as I'm concerned for 1.2 (which isn't the sole consideration, just sharing
> the view from my "station"). I'm happy to leave this here and harmonize the
> interface (e.g. adding missing methods). Or I'm happy to move this anywhere
> else!
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/21907/files#r54664583>.
>